metricflow/model/semantics/metric_lookup.py

```diff
@@ -17,13 +17,12 @@
 )
 from metricflow.model.semantics.semantic_model_join_evaluator import MAX_JOIN_HOPS
 from metricflow.model.semantics.semantic_model_lookup import SemanticModelLookup
-from metricflow.protocols.semantics import MetricAccessor
 from metricflow.specs.specs import TimeDimensionSpec
 
 logger = logging.getLogger(__name__)
 
 
-class MetricLookup(MetricAccessor):
+class MetricLookup:
```

Comment:
I think the `add_metric` method here is the only "write" method in this class, and it's not called anywhere. Mind just renaming it to `_add_metric` so we don't have any "write" methods in the public interface?